packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/entity_type/read.rs

```diff
@@ -0,0 +1,142 @@
+                                        context
+                                            .read_versioned_property_type(property_type_ref.uri())
+                                            .await
+                                            .change_context(ResolveError::StoreReadError)?
+                                            .resolve(property_type_segments, context)
+                                            .await
+                                    })
+                                    .try_collect()
+                                    .await?,
+                            ));
+                        }
+                        [property_type_segment, ..] if property_type_segment.identifier == "*" => {
+                            todo!("{}", UNIMPLEMENTED_WILDCARDS)
+                        }
+                        _ => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                    },
+                    "required" => Literal::List(
+                        self.record
+                            .required()
+                            .iter()
+                            .map(|base_uri| Literal::String(base_uri.to_string()))
+                            .collect(),
+                    ),
+                    "links" => match segments {
+                        [] => todo!("{}", UNIMPLEMENTED_LITERAL_OBJECT),
+                        [entity_type_segment, entity_type_segments @ ..] => {
+                            match entity_type_segment.identifier.as_str() {
+                                "*" => {
+                                    // TODO: Use relation tables,
+                                    //   see https://app.asana.com/0/0/1202884883200942/f
+                                    return Ok(Literal::List(
+                                        stream::iter(self.record.link_type_references())
+                                            .then(|(_, entity_type_ref)| async {
+                                                resolve_entity_type(
+                                                    entity_type_ref.uri(),
+                                                    entity_type_segments,
+                                                    context,
+                                                )
+                                                .await
+                                            })
+                                            .try_collect()
+                                            .await?,
+                                    ));
+                                }
+                                link_type_uri => {
+                                    let versioned_uri = VersionedUri::from_str(link_type_uri)
+                                        .into_report()
+                                        .change_context(ResolveError::StoreReadError)?;
+                                    if let Some(entity_type_ref) =
```

Comment:
Sorry to be pedantic about variable names in this PR, but I think it does help to get these names right. 
Wouldn't the `entity_type_ref` be the `entity_type_ref.uri()` rather than the result of `self.record.link_type_references().get(&versioned_uri)`? As in this is more like the `link_type_ref` or something

Reply:
This should be fine as `link_type_references()` returns a map of a link type URI -> `EntityTypeReference`. From only looking at the GH diff, this is hard to see, but with inline hints, this is more visible:
<img width="590" alt="image" src="https://user-images.githubusercontent.com/21277928/188139046-c4388cea-b70d-4f4c-b176-7288bf07f21a.png">

Does this solve your concern?

Btw. no reason to excuse, good names are hard but important 🙂

Reply:
Yes, thank you!